org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/frameworkevent/FrameworkEventConditions.java

```diff
@@ -0,0 +1,335 @@
+	 *
+	 * public static void example_matches(int typeMask, Class<? extends Throwable> throwableClass) {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(matches(typeMask, throwableClass))
+	 * 		.filteredOn(matches(typeMask, throwableClass))
+	 * 		.first()// map to
+	 * 				// {@link
+	 * 				// ObjectAssert}
+	 * 		.has(matches(typeMask, throwableClass));// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @param eventTypeMask - the typeMask that would be checked against the
+	 *            bundle type of the {@link FrameworkEvent}
+	 * @param throwableClass - the Class that would be checked against the
+	 *            throwable of the {@link FrameworkEvent}
+	 * @return the Condition
+	 */
+	public static Condition<FrameworkEvent> matches(final int eventTypeMask, final Class<? extends Throwable> throwableClass) {
+		return Assertions.allOf(type(eventTypeMask), throwableOfClass(throwableClass));
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a throwable is not null.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_throwableIsNotNull() {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(throwableIsNotNull())
+	 * 		.filteredOn(throwableIsNotNull())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(throwableIsNotNull());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	public static Condition<FrameworkEvent> throwableIsNotNull() {
+		return not(throwableIsNull());
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a throwable is null.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_throwableIsNull() {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(throwableIsNull())
+	 * 		.filteredOn(throwableIsNull())
+	 * 		.first()// map to {@link ObjectAssert}
+	 * 		.has(throwableIsNull());// used on {@link
+	 * 	// ObjectAssert}
+	 * }
+	 * </pre>
+	 *
+	 * @return the Condition
+	 */
+	public static Condition<FrameworkEvent> throwableIsNull() {
+		return VerboseCondition.verboseCondition((fe) -> fe.getThrowable() == null, "throwable is <null>",
+			(fe) -> fe.getThrowable()
+				.toString());
+	}
+
+	/**
+	 * Creates a {@link Condition} to be met by an {@link FrameworkEvent}.
+	 * Checking if a throwable is an instanceof the throwableClass.
+	 *
+	 * <pre>
+	 * List<FrameworkEvent> frameworkEvents = null;
+	 *
+	 * public static void example_throwableOfClass(Class<? extends Throwable> throwableClass) {
+	 *
+	 * 	assertThat(frameworkEvents)// created an {@link ListAssert}
+	 * 		.have(throwableOfClass(throwableClass))
+	 * 		.filteredOn(throwableOfClass(throwableClass))
+	 * 		.first()// map to
+	 * 				// {@link
+	 * 				// ObjectAssert}
```

Comment:
Some of the formatting in the javadoc comments looks weird. I don't think you want to use `{@link}` in the example code.

Reply:
fixed